src/Transition_Systems-Weak_Transition_Systems.thy

```diff
@@ -284,6 +284,30 @@ proof -
     by simp
 qed
 
+lemma weak_tau_transitions_up_to_weak_bisimilarity_rule:
+  assumes
+    "p \<Rightarrow>\<lparr>\<tau>\<rparr> p'"
+  and
+    "p' \<approx> q"
+  and
+    "q \<Rightarrow>\<lparr>\<tau>\<rparr> q'"
+  and
+    "q' \<approx> p"
+  shows
+    "p \<approx> q"
+unfolding weak_bisimilarity_is_mixed_bisimilarity using assms
+proof (coinduction arbitrary: p q p' q' rule: mixed.symmetric_up_to_rule [where \<F> = "[\<asymp>]"])
+  case (simulation \<alpha> s)
+  from \<open>p \<rightarrow>\<lparr>\<alpha>\<rparr> s\<close> and \<open>q \<Rightarrow>\<lparr>\<tau>\<rparr> q'\<close> and \<open>q' \<approx> p\<close> obtain s' where "q' \<Rightarrow>\<lparr>\<alpha>\<rparr> s'" and "s' \<approx> s"
+    using transition_in_weak_transition_rule
+    by (blast elim: weak.bisimilarity.cases)
+  with \<open>q \<Rightarrow>\<lparr>\<tau>\<rparr> q'\<close> have "q \<Rightarrow>\<lparr>\<alpha>\<rparr> s'"
+    by (auto simp add: relcompp_assoc, blast intro: rtranclp_trans)
+  with \<open>s' \<approx> s\<close> show ?case
+    unfolding weak_bisimilarity_is_mixed_bisimilarity
+    by (fastforce intro: mixed.bisimilarity_symmetry_rule)
+qed (respectful, blast)
```

Comment:
I think there is no point conducting this proof using the mixed transition system, as we don’t need to distinguish transitions according to their actions, where having strong transitions instead of weak ones provides a clear advantage.

Reply:
Fixed in 4ce75039b47c85039e138bff90f9341cd5733908, please verify.

Reply:
Yes, looks better now. :slightly_smiling_face: